packages/ui/src/components/Table/TablePagination.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { ThemeProvider } from '@mui/material';
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { theme } from '../theme/theme';
+import { TablePagination, TablePaginationProps } from './TablePagination';
+
+const Pagination = (props: TablePaginationProps) => (
+    <ThemeProvider theme={theme}>
+        <table>
+            <TablePagination {...props} />
+        </table>
+    </ThemeProvider>
+);
+
+const sampleProps = {
+    itemsCount: 10,
+    currentPage: 0,
+    rowsPerPage: 2,
+    onChangePage: () => {},
+};
+
+const evenPages = [0, 1, 2, 3, 4];
+const unevenPages = [0, 1, 2, 3];
+
+describe('<TablePagination />', () => {
+    it('renders the correct number of pages', () => {
+        const { getByRole } = render(<Pagination {...sampleProps} />);
+        evenPages.forEach((page) => expect(getByRole('button', { name: `${page + 1}` })).toBeInTheDocument());
+    });
+
+    it("renders the correct number of pages when the items doesn't divide exactly between pages", () => {
+        const { getByRole } = render(
+            <Pagination currentPage={0} itemsCount={10} rowsPerPage={3} onChangePage={() => {}} />
+        );
+
+        unevenPages.forEach((page) => expect(getByRole('button', { name: `${page + 1}` })).toBeInTheDocument());
+    });
+
+    it('renders the prev and next buttons', () => {
+        const { getByTestId } = render(
+            <Pagination currentPage={0} itemsCount={10} rowsPerPage={2} onChangePage={() => {}} />
+        );
+
+        expect(getByTestId('table-pagination-item-next')).toBeInTheDocument();
```

Comment:
it's better to make sure that the buttons have proper aria labels such as 'previous page' and 'next page' and use `getByRole` instead. This way we also handle accessibility for the component. Same for the other case where we have `getByTestId`